src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -358,10 +359,36 @@ private bool BeforeGraphResolutionValidations(int httpSourcesCount)
             success &= EvaluateHttpSourceUsage();
             success &= HasValidPlatformVersions();
             success &= PackageReferencesHaveVersions();
+            success &= ValidatePackageReferences();
 
             return success;
         }
 
+        private bool ValidatePackageReferences()
+        {
+            foreach (var frameworkInfo in _request.Project.TargetFrameworks)
+            {
+                foreach (var dependency in frameworkInfo.Dependencies)
+                {
+                    try
+                    {
+                        PackageIdValidator.ValidatePackageId(dependency.Name);
+                    }
+                    catch (ArgumentException)
+                    {
+                        _logger.Log(
+                            RestoreLogMessage.CreateError(
+                            NuGetLogCode.NU1017,
+                            string.Format(CultureInfo.CurrentCulture, Strings.Error_invalid_packageid, dependency.Name)));
```

Comment:
I think we should use `ArgumentException.Message` because the [exception raised by `ValidatePackageId`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/PackageCreation/Utility/PackageIdValidator.cs#L31-L42) because the message is actionable.